### ISSUE TITLE
ONNX: Models with InputValue Shapes that are defined via `DimParam` (String) can't be run

### ISSUE THREAD

**Issue by LLukas22** [huggingface/candle#1281]
Some ONNX models (e.g. transfomerers like [RoBERTa](https://github.com/onnx/models/blob/main/text/machine_comprehension/roberta/model/roberta-base-11.onnx)) don't define their input shapes as `DimValue` (int) but as a `DimParam` (string) e.g. instead of `(1, 512)` they define thier input shape as `("batch_size", "seq_len")`.

Currently the validation step in [`simple_eval`](https://github.com/huggingface/candle/blob/main/candle-onnx/src/eval.rs#L176-L184) doesn't allow to run such models. But even after removing the validation step the inference runs into some sort of error with missmatching shapes: `"unsupported op_type Shape for op NodeProto { input: [\"input_ids\"], output: [\"200\"], name: \"Shape_0\", op_type: \"Shape\", domain: \"\", attribute: [], doc_string: \"\" }"` 

This probably means that there is some additional logic needed to handle these "dynamic"/"undefined" input shapes

**Comment by LaurentMazare** [comment]
I've merged support for a couple more ops that are required for this model, some of them in a hacky way (e.g. 'gather') as they are hard to support with the current candle ops. There are still some missing ops, the first one is `CumSum` which we don't support well at the moment - I'll push a bit more on this when I get a chance, probably by just handling `CumSum` in a non differentiable way via a custom-op or something like this.

**Comment by LLukas22** [comment]
Yup, thanks for adding those ops. I looked at some more onnx models and nearly all transformer based models seam to utalize `CumSum` at the beginning of the model, probably to create some sort of mask. I first thought that this was some sort of legacy behaviour of onnx but even the newest onnx models, created via the newest optimum and pytorch version utalize `CumSum`. 

 If i find some time i could try to add `CumSum`, but i have no idea how i would go about implementing it in a fast and good way 🤔

**Comment by xd009642** [comment]
Just chiming in to say I'm facing this myself with a TTS model I'm trying out, a batch size parameter and a sequence one. If there's any way to help I'd be willing to assist in terms of a PR etc - or just sharing the neural network if it will help.

```
Graph input: ValueInfoProto { name: "phonemes", r#type: Some(TypeProto { denotation: "", value: Some(TensorType(Tensor { elem_type: 7, shape: Some(TensorShapeProto { dim: [Dimension { denotation: "", value: Some(DimParam("batch_size")) }, Dimension { denotation: "", value: Some(DimParam("sequence")) }] }) })) }), doc_string: "" }

Graph input: ValueInfoProto { name: "plen", r#type: Some(TypeProto { denotation: "", value: Some(TensorType(Tensor { elem_type: 7, shape: Some(TensorShapeProto { dim: [Dimension { denotation: "", value: Some(DimParam("batch_size")) }] }) })) }), doc_string: "" }
```
